### PR TITLE
feat: add support for children transclude

### LINKED ISSUE #12: Add support for children/transclusion
See https://github.com/coatue-oss/react2angular/issues/11

### PR DESCRIPTION
**Pull request by spirosikmd**
Fixes #12 

@bcherny I followed the recommendations mentioned in the issue. I created a prototype. Seems that it is possible. However, the last part I am not sure how to proceed best. Current status is:

```
<test-angular-two foo="foo" bar="bar" baz="baz" class="ng-scope ng-isolate-scope">
  <div data-reactroot="">
    <p><!-- react-text: 3 -->Foo: <!-- /react-text --><!-- react-text: 4 -->1<!-- /react-text --></p>
    <p><!-- react-text: 6 -->Bar: <!-- /react-text --><!-- react-text: 7 -->true,false<!-- /react-text --></p>
    <p>Baz</p>
    <!-- react-text: 9 -->&lt;span className="ng-scope"&gt;Transcluded&lt;/span&gt;<!-- /react-text -->
    <!-- react-text: 10 -->&lt;p className="ng-scope"&gt;Transcluded&lt;/p&gt;<!-- /react-text -->
  </div>
</test-angular-two>
```

the transcluded components are added as children, but they are escaped. Any idea? Maybe use [angular $sce](https://docs.angularjs.org/api/ng/service/$sce)?

### CODE REVIEW

**Comment by spirosikmd** [comment]
So @bcherny I managed to fix the escaping issue by using [`html-react-parser`](https://www.npmjs.com/package/html-react-parser)! I think this is ready for review. Let me know what you guys think?

**Comment by bcherny** [comment]
@spirosikmd This is excellent, thanks for that hard work! A couple of notes:

- Can you please add tests for updating the scope when transcluded elements use stuff from that scope (making sure the transcluded content updates, isn't double-rendered, etc.)
- How much do the extra dependencies add to the build size?

**Comment by spirosikmd** [comment]
Hi @bcherny! Thank you for providing the package in the first place!

The sizes for `support-transclude` branch are:
```
577B index.d.ts
2.0K index.es2015.js
2.3K index.js
1.5K index.js.map
```

compared to `master` branch:
```
577B index.d.ts
1.4K index.es2015.js
2.4K index.js
889B index.js.map
```

Size of `index.js` hasn't increased at all. However, `index.es2015.js` size increased by ~0.6K. Do you think this is would be a problem?

**Comment by bcherny** [comment]
@spirosikmd 0.6kb is no big deal, but what happens when you run that through browserify (to pull in  htmltojsx, html-react-parser, and all of their dependencies)?

**Comment by spirosikmd** [comment]
When running the following command

```
./node_modules/.bin/browserify index.js -o bundle.js
```

on both branches, the `bundle.js` file ends up being 2.5M for `support-transclude` and 1.1M for `master`. Big difference I think. How to improve that?

**Comment by bcherny** [comment]
2.5mb is fine I think. A tree-shaking compiler (rollup, webpack) will improve that size, and we get a lot of functionality for that extra 1.4mb.

**Comment by spirosikmd** [comment]
Cool ok then! I will work on adding some tests and docs for this feature.

File: test.tsx [hunk 1]
```diff
@@ -182,17 +182,47 @@ describe('react2angular', () => {
...
+    it('should update with children', () => {
+      const scope = Object.assign($rootScope.$new(true), {
+        bar: [true, false],
+        foo: 1
+      })
+      const element = $(`<test-angular-one foo="foo" bar="bar"><span>{{foo}}</span></test-angular-one>`)
+      $compile(element)(scope)
+      $rootScope.$apply()
+      expect(element.find('span').text()).toBe('1')
+      // scope.$apply(() =>
```
Comment (spirosikmd) [inline test.tsx]: @bcherny we can now transclude other components as well, and bind once with initial rendering. However, I cannot manage to update the two way binding. Any idea? Would it be possible to release without this support for now?

Comment (bcherny) [reply test.tsx]: @spirosikmd This looks like a pretty serious bug to me. In your test it's not that the view doesn't update, it's that it reverts to its original untemplated version (eg. in this case it's `{'{'}{'{'}foo{'}'}{'}'}`. If that's correct, I would argue this should be fixed before we merge. Broken support for transclusion (that sets false expectations) could be worse than no support.

Comment (spirosikmd) [reply test.tsx]: You are definitely right. This should be fixed first!

**Comment by bcherny** [comment]
@spirosikmd Any updates on this?

**Comment by spirosikmd** [comment]
Hey @bcherny! I haven't worked on this yet. I am trying the approach described in [this article](https://medium.com/@jrwebdev/migrating-an-angular-1-application-to-react-8891ec73d462), passing all the data needed as props and don't use transclusion.

**Comment by bcherny** [comment]
@spirosikmd Sounds good! I was doing some PR cleanup and wanted to make sure this wasn't forgotten.

**Comment by andreoav** [comment]
Any update?

**Comment by gairon** [comment]
Any update?

**Comment by spirosikmd** [comment]
Hi @andreoav @gairon! I stopped working on this. I won't invest in angular.js anymore. I see @gregbty is working on a similar PR. He runs into the same issues I was running as well.

**Pull request closed without merging**